File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int,int>mp;
        for(int i=0;i<nums.size();i++){
            int a=nums[i];
            int val=target-a;
            if(mp.find(val)!=mp.end()){
                return {mp[val],i};
            }
            mp[a]=i;
        }
        return {-1,-1};  
    }
};

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> h = new HashMap<>();
        for(int i=0;i<nums.length;i++){
            int a=nums[i];
            int more=target-a;
             if (h.containsKey(more)) { 
                return new int[]{h.get(more), i}; 
            }
            h.put(a, i); 
        }
        return new int[]{-1, -1}; 
    }
}